src/Compatibility/Microsoft.DotNet.ApiCompatibility/Microsoft.DotNet.ApiCompatibility.csproj

```diff
@@ -0,0 +1,18 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
```

Comment:
I suspect you'll need to run on .NETFramework in the context of the desktop msbuild and may want to explicitly target it.   Follow the pattern used by other SDK assemblies.

Reply:
It seems like other SDK assemblies target net472 and/or net6.0. However I did find a source generator that targets netstandard2.0. So if you have the preference of targeting net472 and netstandard2.0 or netcoreapp3.1 I can do that. Whatever your preference is.

Reply:
It’s not my preference that matters, you’ll need to consider the implications. 

For source generators we have a host that ensures shims are present along with any necessary bindingRedirects.  In general analyzers/source generators don’t need to carry dependencies since they typically just rely on the compiler platform assemblies.

MSBuild tasks on the other hand don’t get much from the host other than the MSBuild interfaces and need to carry their dependencies. Our versioning policy for some dotnet/runtime assemblies will result in versioning of netframework assemblies and not netstandard nor netcore.  As a result we may introduce the need for more bindingRedirects if we don’t target netframework.  As you know bindingRedirects are difficult to deal with in MSBuild tasks (we should avoid AssemblyRrsolve hack in production). I believe the SDK assemblies take special care to avoid them.

It’s fine to defer until it’s a problem if you want.  I was expecting there was an established pattern in the SDK.

Reply:
Thanks for elaborating. It seems like the pattern they have or the way the avoid this is they target net472;net6.0 (or whatever the current tfm is), so I think we could safely target `net472;netcoreapp3.1` here? Or would we need to include a `netstandard2.0` asset as well?